### ISSUE TITLE
reserve a test

### ISSUE THREAD

**Issue by chrishat34** [2e0byo/YADC#3]
hi 2e0byo
On line 208 in scraper, you make these remarks:
```python
        # could improve here:
        # scroll to the right calendar month
        # click
        # reserve test
```
I would be really grateful if you could make that happen please. 
I have managed to click the 1st date but cannot seen to find a way to click the 1st time(on that date)
regards

**Comment by 2e0byo** [comment]
Does that mean you managed to find a test?  Anyhow, the problem is that I don't have any tests to test it on.  What did you try to click the first date?  Have a look at one of the linked projects in the Readme---one of them had auto booking.  I'll have a look when I get back, but since I can't test it I'll be implementing blind, which is pretty dangerous....

**Comment by chrishat34** [comment]
Hi, yes found some dates and just wanted to hold it for the 15 or so
mins whilst i rush to get to the pc. ive had a good old look at the code
you refer to but im just not skilled enough to bring the solution into your
code. Im happy to try out any code snippets you might be able to put
together.
regards

**Comment by chrishat34** [comment]
....so not looking to auto book, just get to the point it is held for me and give me extra time to check its good for me then swap the dates

**Comment by 2e0byo** [comment]
This is likely to have to wait until the weekend.

**Comment by chrishat34** [comment]
That world be brilliant. Thank you.

**Comment by 2e0byo** [comment]
See the draft implementation on the new `booking` branch in 2ed7468.

This is completely untested, but it should hopefully be enough to get things going.  I have taken the logic directly from the code which inspired this repo (see the readme for a link).  Note two new params passed when instantising a `Scraper()`:

- `reserve` : default= True: reserve tests.
- `short_notice`: default = True: reserve short notice tests.  Note that I've a feeling these can't be cancelled or something: worth checking.

The implementation will likely need tweaking.  I left out all attempts at error checking, so if the underlying `Browser()` doesn't manage to solve the captcha for you (which it will try to do, I think---but it was a while ago I wrote it and I can't quite remember which calls trigger a check for catpchas....) it will simple throw an error.  Since error behaviour is really nice (you get a screenshot  + traceback + page source in the errors dir) you should be able to figure out what's happening and add the required code fairly easily.

Good luck!

**Comment by chrishat34** [comment]
You did amazingly given no testing. Did a few tweaks and now have a version of it running nicely - thank you, i was so close. I'll have to have a good look at the error behaviour - the screenshots etc,  as i have never had that running even in the previous code. I suspect something to do with linux v windows configuration

**Comment by 2e0byo** [comment]
Great: if you open a pr with your working version I'll merge it.

Errors should land in the `errors` dir you pass when instantiating the
scraper.

Le dim. 12 déc. 2021 à 11:42, chrishat34 ***@***.***> a
écrit :

**Comment by 2e0byo** [comment]
any chance you could open a PR with your changes? I'll happily help cleaning it up if need be.  (Once you push your changes up you should be able to open the PR from the main github page for your repo.)

**Comment by 2e0byo** [comment]
closed in #5.
